### PR TITLE
fix: physcis event remove when collider shape destroy

### PR DESCRIPTION
**Pull request by yangfengzzz**
### Please check if the PR fulfills these requirements

- [ ] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### Other information:

### CODE REVIEW

File: packages/physics-lite/src/LitePhysicsManager.ts [hunk 1]
```diff
@@ -65,6 +65,16 @@ export class LitePhysicsManager implements IPhysicsManager {
    * {@inheritDoc IPhysicsManager.removeColliderShape }
    */
   removeColliderShape(colliderShape: LiteColliderShape): void {
+    const { _eventPool: eventPool, _currentEvents: currentEvents } = this;
+    for (let i = currentEvents.length - 1; i >= 0; i--) {
+      const event = currentEvents.get(i);
+      if (event.index1 == colliderShape._id || event.index2 == colliderShape._id) {
+        event.state = TriggerEventState.Exit;
```
Comment (cptbtptpbcptdtptp) [inline packages/physics-lite/src/LitePhysicsManager.ts:72]: Unnecessary.

**Review by cptbtptpbcptdtptp — CHANGES_REQUESTED** [review]

File: packages/physics-lite/src/LitePhysicsManager.ts [hunk 2]
```diff
@@ -65,6 +65,16 @@ export class LitePhysicsManager implements IPhysicsManager {
...
+        currentEvents.deleteByIndex(i);
+        eventPool.push(event);
+      }
+    }
+
```
Comment (cptbtptpbcptdtptp) [inline packages/physics-lite/src/LitePhysicsManager.ts:77]:
``` 
  const { _eventPool: eventPool, _currentEvents: currentEvents } = this;
  const { _id: shapeID } = colliderShape;
  for (let i = currentEvents.length - 1; i >= 0; i--) {
    const event = currentEvents.get(i);
    if (event.index1 === shapeID || event.index2 === shapeID) {
      currentEvents.deleteByIndex(i);
      eventPool.push(event);
    }
  }
  delete this._eventMap[shapeID];
```

File: packages/physics-physx/src/PhysXPhysicsManager.ts [hunk 3]
```diff
@@ -117,6 +117,16 @@ export class PhysXPhysicsManager implements IPhysicsManager {
...
+    const { _eventPool: eventPool, _currentEvents: currentEvents } = this;
+    for (let i = currentEvents.length - 1; i >= 0; i--) {
+      const event = currentEvents.get(i);
+      if (event.index1 == colliderShape._id || event.index2 == colliderShape._id) {
+        event.state = TriggerEventState.Exit;
+        currentEvents.deleteByIndex(i);
+        eventPool.push(event);
+      }
+    }
+
```
Comment (cptbtptpbcptdtptp) [inline packages/physics-physx/src/PhysXPhysicsManager.ts:129]:
```
  const { _eventPool: eventPool, _currentEvents: currentEvents } = this;
  const { _id: shapeID } = colliderShape;
  for (let i = currentEvents.length - 1; i >= 0; i--) {
    const event = currentEvents.get(i);
    if (event.index1 === shapeID || event.index2 === shapeID) {
      currentEvents.deleteByIndex(i);
      eventPool.push(event);
    }
  }
  delete this._eventMap[shapeID];
```

**Review by cptbtptpbcptdtptp — CHANGES_REQUESTED** [review]

File: packages/physics-lite/src/LitePhysicsManager.ts (hunk 2 above)
Comment (yangfengzzz) [reply packages/physics-lite/src/LitePhysicsManager.ts]: done

File: packages/physics-physx/src/PhysXPhysicsManager.ts (hunk 3 above)
Comment (yangfengzzz) [reply packages/physics-physx/src/PhysXPhysicsManager.ts]: done

**Review by GuoLei1990 — APPROVED** [review]
+1

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

**Pull request merged**